File: concurrency/ruby/hello1.rb
count = 0
threads = []

10.times do |i|
  threads << Thread.new do
    greeting_message = "Hello"

    # This is weird in Ruby but it's closer to the println! macro
    # usage in the Rust example.
    puts "#{greeting_message} from lightweight thread number #{i}\n"

    count = count + 1
  end
end

# wait for all thread finishes
threads.each &:join

File: concurrency/rust/hello1.rs
/* This function creates ten "tasks" that all execute concurrently.
   To verify this, run the program several times and observe the irregular
   order in which each task's output is printed. */
fn main() {
    // This string is immutable, so it can safely be accessed from multiple tasks.
    let greeting = "Hello";
    // `for` loops work with any type that implements the `Iterator` trait.
    for num in range(0i, 10) {
        spawn(proc() {
            // `println!` is a macro that statically typechecks a format string.
            // Macros are structural (as in Scheme) rather than textual (as in C).
            println!("{:s} from lightweight thread number {:i}", greeting, num);
        });
    }
}